packages/extension/src/utils/getDocumentCookies.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { type Cookie as ParsedCookie } from 'simple-cookie';
+import { isFirstParty } from '@cookie-analysis-tool/common';
+
+/**
+ * Internal dependencies.
+ */
+import { createCookieObject } from '../worker/createCookieObject';
+import { fetchDictionary } from './fetchCookieDictionary';
+import findAnalyticsMatch from '../worker/findAnalyticsMatch';
+import { CookieStore, type CookieData } from '../localStore';
+
+/**
+ * Adds the cookies set via document.cookie.
+ * @param tabId Id of the tab being listened to.
+ */
+async function getDocumentCookies(tabId: string) {
+  let documentCookies: string[] = [];
+  let parsedCookieData: CookieData[] = [];
+  let tabUrl = '';
+
+  const dictionary = await fetchDictionary();
+
+  if (!tabId || tabId === '') {
+    return;
+  }
+
+  chrome.devtools.inspectedWindow.eval(
+    'document.cookie.split(";")',
+    (result: string[], isException) => {
+      if (!isException && result.length > 1) {
+        documentCookies = result;
+
+        chrome.devtools.inspectedWindow.eval(
+          'window.location.href',
+          async (tabUrlResult: string, tabUrlIsException) => {
+            if (!tabUrlIsException && typeof tabUrlResult === 'string') {
+              tabUrl = tabUrlResult;
+
+              const frames = await chrome.webNavigation.getAllFrames({
+                tabId: Number(tabId),
+              });
+
+              const outerMostFrame = frames?.filter(
+                (frame) => frame.frameType === 'outermost_frame'
+              );
+
+              parsedCookieData = await Promise.all(
+                documentCookies.map(async (singleCookie) => {
+                  const cookieValue = singleCookie.split('=')[1]?.trim();
+                  const cookieName = singleCookie.split('=')[0]?.trim();
+                  let analytics;
+
+                  let parsedCookie = {
+                    name: cookieName,
+                    value: cookieValue,
+                  } as ParsedCookie;
+
+                  parsedCookie = await createCookieObject(parsedCookie, tabUrl);
```

Comment:
Overriding the same variable looks a bit weird here. Either use a different name or pass the object directly in `createCookieObject`

Reply:
Addressed in [46de8a2](https://github.com/GoogleChromeLabs/ps-analysis-tool/pull/146/commits/46de8a2e33ccba5498b6ca787b73200ae569a80e)